fedb/api-server.md

```diff
@@ -0,0 +1,178 @@
+- Start Date: 2021-04-23
+- Target Major Version: (2.2.0)
+- Reference Issues: 
+- Implementation PR: 
+
+# Add API Server
+
+# Summary
+
+HTTP interface is one of the most friendly interface and many developers like using http interface to develop their program. REST API also help us to show demo simply. So we want to support uing http api to access FEDB.
+
+# Detailed Design
+
+## Interface Design
+
+### Schema Mapping
+|FEDB type|Json type|
+|---|---|
+|string|String|
+|smallint|Number|
+|int|Number|
+|bigint|Number|
+|float|Number|
+|double|Number|
+|bool|Boolean|
+|null|null|
+|date|String|
+|timestamp|String|
+
+JSON does not have a built-in type for date/timestamp values. We store the date/timestamp value as a string in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format, e.g 2021-05-06, 2021-05-06T00:02:32Z.
+
+### HTTP Status 
+|code|message|
+|--|--|
+|200|ok|
+|404|page not found|
+|408|request timeout|
+
+### Put
+reqeust url: http://ip:port/dbs/{db_name}/tables/{table_name}  
+http method: PUT  
+request body: 
+```
+{
+    "value": [{
+        "field1": "value1",
+        "field2": 111,
+        "field3": 1.4,
+        "field4": "2021-04-27",
+        "field5": "2021-04-27T08:03:15+00:00",
+        "field6": true
+    }]
+
+}
+```
+response:
+```
+{
+    "code": 0,
+    "msg": "ok"
+}
+```
+
+|code|message|
+|--|--|
+|0|ok|
+|1|get insert information failed|
+
+### Execute Procedure 
+reqeust url: http://ip:port/dbs/{db_name}/procedures/{procedure_name}  
+http method: POST  
+request body: 
+```
+{
+    "common_cols":["value1", "value2"],
+    "input": [["value0", "value3"],["value0", "value3"]],
+    "need_schema": true
+}
+```
+response:
+```
+{
+    "code": 0,
+    "msg": "ok",
+    "data": {
+        "schema": [{"field1":"bool"}, {"field2":"int32"}],
+        "data": [["value1", "value2"], [...]]
```

Comment:
The `response` should include common_cols and  none_common_cols

Reply:
ok